### ISSUE TITLE
Add support for wayland-wall protocols

### ISSUE THREAD

**Issue by lf-** [SirCmpwn/sway#1275]
wayland-wall is a bunch of protocols for window-manager-only setups. It includes such things as allowing a client to dock windows on screen edges (`dock-manager`), ask for keyboard focus such as for launchers (`launcher-menu`), or see, control, and focus other clients' surfaces (`window-switcher`).

At the very least, the launcher-menu protocol would be tremendously helpful since rofi is currently unusable due to it being unable to grab keyboard focus. Currently it is possible to hack wlc into supporting these by using LD_PRELOAD (which is a non-solution).

Spec: https://github.com/wayland-wall/wayland-wall

Implementation providing horrible LD_PRELOAD hack: https://github.com/wayland-wall/orbment-wall

**Comment by SirCmpwn** [comment]
I don't intend to support wayland-wall. We have a better strategy via [sway-protocols](https://github.com/SirCmpwn/sway-protocols) that we'll use to support more use-cases more elegantly when the time comes.

**Comment by lf-** [comment]
This seems to imply that they're not implemented yet, is that correct?

**Comment by SirCmpwn** [comment]
Correct.

**Comment by lf-** [comment]
Well, that represents a decently large gap in functionality. I rely on rofi
every day to launch applications and it effectively isn't any good on sway
since it doesn't focus automatically (though it might on some other
implementations that have wayland-wall).

Is there any way of showing a dmenu like dialog with sway then? Or do I
need to hack something together using the IPC interface to focus any rofi
window as soon as it's started?

**Comment by SirCmpwn** [comment]
dmenu itself works fine on sway.

**Comment by tmccombs** [comment]
I much prefer rofi to dmenu. Will wayland-native rofi (which as I understand it will use the wayland-wall protocol) work on sway?

**Comment by lf-** [comment]
Kind of. My friend who uses sway all the time added a rule to just make
rofi always on top, which seems to do the trick. Ask @parrottq if you need
more details.

**Comment by SirCmpwn** [comment]
If it's going to use wayland-wall, no. Read the issue for our plans and speak with rofi if you are concerned about their support - we are are not rofi.
